src/EntryPoints/Specials/SpecialManageApprovers.php

```diff
@@ -109,10 +107,19 @@ private function approversToViewModel( array $approvers ): array {
 	 * @return array<string, mixed>
 	 */
 	private function approverToViewModel( Approver $approver ): array {
+		$showCategories = true;
+
+		if ( empty( $approver->categories ) ) {
+			$request = $this->getRequest();
+			$user = $this->userFactory->newFromName( $request->getText( 'username' ) );
+			$showCategories = $request->wasPosted() && $user->getName() === $approver->username;
+		}
```

Comment:
This can be in its own function returning a bool

And you can do it without mutation (assigning to an already assigned variable)

Reply:
Took me a bit to understand why we have this logic at all. It's because using the UI causes post requests which reload the page, so without this logic, the row would vanish as soon as you delete the last category, correct? 

(Having a well names private function would make this more obvious)

Reply:
Yes. That's right.